EventListener/RateLimitAnnotationListener.php

```diff
@@ -163,48 +160,16 @@ protected function findBestMethodMatch(Request $request, array $annotations)
         return $best_match;
     }
 
-    private function getKey(FilterControllerEvent $event, RateLimit $rateLimit, array $annotations)
+    private function getKey(LimitProcessorInterface $limitProcessor, RateLimit $rateLimit, Request $request)
```

Comment:
Should getKey also be in the Manager/Service?

Reply:
As I see, existing RateLimitService class stable, key generation logic not. 
If we put  getKey into RateLimitService class, extension will be needed, from another side in Laravel I don't have symfony events logic, so didn't see the reason to move all this method or some part of it into Service class.

Logic `$limitProcessor->getRateLimitAlias($request) . '. ' . $limitProcessor->getRateLimitAlias($request)`  can be moved just in separate class, but it will be created directly inside Listener or iit will have 3 method params
```
class RateLimitBaseKey
{
    /**
     * @var RateLimit
     */
    private $rateLimit;

    /**
     * @var PathLimitProcessor
     */
    private $limitProcessor;

    public function __construct(RateLimit $rateLimit, PathLimitProcessor $limitProcessor)
    {
        $this->rateLimit = $rateLimit;
        $this->limitProcessor = $limitProcessor;
    }

    public function getKey(Request $request)
    {
        return trim(
            join('.', $this->rateLimit->getMethods()) . '.' . $this->limitProcessor->getRateLimitAlias($request),
            '.'
        );
    }
}
```